packages/suite/src/utils/wallet/solanaUtils.ts

```diff
@@ -0,0 +1,260 @@
+import * as BufferLayout from '@solana/buffer-layout';
+import type { TokenAccount } from '@trezor/blockchain-link-types';
+import { A, F, pipe } from '@mobily/ts-belt';
+import { getLamportsFromSol } from '@suite-common/wallet-utils';
+import {
+    TOKEN_PROGRAM_PUBLIC_KEY,
+    ASSOCIATED_TOKEN_PROGRAM_PUBLIC_KEY,
+    SYSTEM_PROGRAM_PUBLIC_KEY,
+} from '@trezor/blockchain-link-utils/lib/solana';
+import BigNumber from 'bignumber.js';
+
+const loadSolanaLib = async () => {
+    const lib = await import('@solana/web3.js');
+
+    return lib;
+};
+
+export const getPubKeyFromAddress = async (address: string) => {
+    const { PublicKey } = await loadSolanaLib();
+
+    return new PublicKey(address);
+};
+
+// To transfer Solana tokens we need to manually encode a buffer with the correct layout
+// according to the instruction standard.
+const LAYOUT = BufferLayout.union(BufferLayout.u8('instruction'));
+LAYOUT.addVariant(
+    12,
+    BufferLayout.struct([BufferLayout.nu64('amount'), BufferLayout.u8('decimals')]),
+    'transferChecked',
+);
+
+const instructionMaxSpan = Math.max(...Object.values(LAYOUT.registry).map((r: any) => r.span));
```

Comment:
Why is this defined outside `encodeTokenTransferInstructionData` function?

Reply:
Good catch, moving it inside the function.